tls/s2n_record_write.c

```diff
@@ -111,10 +105,18 @@ int s2n_record_min_write_payload_size(struct s2n_connection *conn)
         size -= 1;
     }
 
-    ENSURE_POSIX(size > 0, S2N_ERR_FRAGMENT_LENGTH_TOO_SMALL);
-    ENSURE_POSIX(size <= ETH_MTU, S2N_ERR_FRAGMENT_LENGTH_TOO_LARGE);
+    /* subtract overheads of a TLS record */
+    uint16_t overhead = 0;
+    GUARD_RESULT(s2n_tls_record_overhead(conn, &overhead));
+    ENSURE(size > overhead, S2N_ERR_FRAGMENT_LENGTH_TOO_SMALL);
+    size -= overhead;
+
+    ENSURE(size > 0, S2N_ERR_FRAGMENT_LENGTH_TOO_SMALL);
+    ENSURE(size <= ETH_MTU, S2N_ERR_FRAGMENT_LENGTH_TOO_LARGE);
 
-    return size;
+    *payload_size = size;
```

Comment:
Can this overflow?

Reply:
it shouldn't. the check before this checks that